sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public T this[int index] { get { throw null; } set { } }
+        public void Add(T item) { }
+        public void Clear() { }
+        public bool Contains(T item) { throw null; }
+        public void CopyTo(T[] array, int arrayIndex) { }
+        public System.Collections.Generic.IEnumerator<T> GetEnumerator() { throw null; }
+        public int IndexOf(T item) { throw null; }
+        public void Insert(int index, T item) { }
+        public bool Remove(T item) { throw null; }
+        public void RemoveAt(int index) { }
+        public void Reset() { }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public static partial class OptionalProperty
+    {
+        public static bool IsCollectionDefined<T>(System.Collections.Generic.IEnumerable<T> collection) { throw null; }
+        public static bool IsCollectionDefined<TKey, TValue>(System.Collections.Generic.IDictionary<TKey, TValue> collection) { throw null; }
+        public static bool IsCollectionDefined<TKey, TValue>(System.Collections.Generic.IReadOnlyDictionary<TKey, TValue> collection) { throw null; }
+        public static bool IsDefined(object value) { throw null; }
+        public static bool IsDefined(string value) { throw null; }
+        public static bool IsDefined(System.Text.Json.JsonElement value) { throw null; }
+        public static bool IsDefined<T>(T? value) where T : struct { throw null; }
+        public static System.Collections.Generic.IDictionary<TKey, TValue> ToDictionary<TKey, TValue>(System.Net.ClientModel.Internal.OptionalProperty<System.Collections.Generic.IDictionary<TKey, TValue>> optional) { throw null; }
+        public static System.Collections.Generic.IReadOnlyDictionary<TKey, TValue> ToDictionary<TKey, TValue>(System.Net.ClientModel.Internal.OptionalProperty<System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>> optional) { throw null; }
+        public static System.Collections.Generic.IList<T> ToList<T>(System.Net.ClientModel.Internal.OptionalProperty<System.Collections.Generic.IList<T>> optional) { throw null; }
+        public static System.Collections.Generic.IReadOnlyList<T> ToList<T>(System.Net.ClientModel.Internal.OptionalProperty<System.Collections.Generic.IReadOnlyList<T>> optional) { throw null; }
+        public static T? ToNullable<T>(System.Net.ClientModel.Internal.OptionalProperty<T?> optional) where T : struct { throw null; }
+        public static T? ToNullable<T>(System.Net.ClientModel.Internal.OptionalProperty<T> optional) where T : struct { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct OptionalProperty<T>
+    {
+        private readonly T _Value_k__BackingField;
+        private readonly int _dummyPrimitive;
+        public OptionalProperty(T value) { throw null; }
+        public bool HasValue { get { throw null; } }
```

Comment:
Why not `IsDefined`? Should all the `Optional*` types share the same naming here?